packages/legend-studio/src/stores/editor-state/element-editor-state/service/ServiceRegistrationState.ts

```diff
@@ -217,19 +231,42 @@ export class ServiceRegistrationState {
           versionInput,
         )) as ServiceRegistrationResult;
       if (this.activatePostRegistration) {
+        this.setRegistrationPhase(
+          SERVICE_REGISTRATION_PHASE.ACTIVATING_SERVICE,
+        );
         yield this.editorStore.graphManagerState.graphManager.activateService(
           serverUrl,
           serviceRegistrationResult.serviceInstanceId,
         );
       }
-      this.setShowModal(false);
-      this.editorStore.applicationStore.notifySuccess(
-        `service with patten ${serviceRegistrationResult.pattern} registered ${
-          this.activatePostRegistration ? ' and activated ' : ''
-        } on ${serviceRegistrationResult.serverURL}`,
-        undefined,
-        null,
-      );
+      const message = `Service with patten ${
+        serviceRegistrationResult.pattern
+      } registered ${this.activatePostRegistration ? 'and activated ' : ''}on ${
```

Comment:
I think we can kill the `on ...` part. `...registered and activated` sounds alright already.